streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -450,26 +450,19 @@ public void register(final StateStore store,
         return (S) stateStores.get(name);
     }
 
+    @SuppressWarnings("deprecation") // removing #schedule(final long intervalMs,...) will fix this
     @Override
-    @Deprecated
-    public Cancellable schedule(final long intervalMs,
+    public Cancellable schedule(final Duration interval,
                                 final PunctuationType type,
-                                final Punctuator callback) {
+                                final Punctuator callback) throws IllegalArgumentException {
+        final long intervalMs = ApiUtils.validateMillisecondDuration(interval, "interval");
```

Comment:
Seem we are missing a check for `intervalMs < 1` -- can we add it?

Reply:
Sure, I'll add them tomorrow morning (CEST)

Reply:
I managed to do the change now, feel free to review it if you have the time.